File: src/AssignCookies.java
// Problem Number: 455
// https://leetcode.com/problems/assign-cookies/

import java.util.Arrays;

public class AssignCookies {
    public int findContentChildren(int[] g, int[] s) {
        Arrays.sort(g);
        Arrays.sort(s);
        int contendedChildren = 0;
        for (int i= 0, j = 0 ; i < g.length && j < s.length ; j++) {
            if (s[j] >= g[i]) {
                contendedChildren++;
                i++;
            }
        }
        return contendedChildren;
    }
}

File: python/assign_cookies.py
from typing import List


class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        s.sort()
        g.sort()
        i = 0
        contendedChildren = 0
        for j in range(len(s)):
            if i < len(g) and s[j] >= g[i]:
                contendedChildren += 1
                i += 1
        return contendedChildren
